### PR TITLE
Add proxy types

### PR DESCRIPTION
**Pull request by cdamian**

### CODE REVIEW

File: gethrpc/client.go [hunk 1]
```diff
@@ -480,17 +480,33 @@ func (c *Client) send(ctx context.Context, op *requestOp, msg interface{}) error
 }
 
 func (c *Client) write(ctx context.Context, msg interface{}) error {
```
Comment (mikiquantum) [inline gethrpc/client.go:482]: 👍

File: types/header.go [hunk 2]
```diff
@@ -69,3 +69,41 @@ func (b *BlockNumber) Decode(decoder scale.Decoder) error {
 	*b = BlockNumber(u.Uint64())
 	return err
 }
+
+type OptionBlockNumber struct {
```
Comment (mikiquantum) [inline types/header.go:73]: Doesnt it make sense to use the Option[BlockNumber] generic here?

File: types/metadataV14.go [hunk 3]
```diff
@@ -160,12 +160,12 @@ func (m *MetadataV14) FindError(moduleIndex U8, errorIndex U8) (*MetadataError,
 				}
 
 				for _, variant := range errType.Def.Variant.Variants {
-					if variant.Index == errorIndex {
+					if variant.Index == errorIndex[0] {
```
Comment (mikiquantum) [inline types/metadataV14.go:163]: If the [0] is the error index in the module/pallet, the what are the other 3 positions for? Looking at the metadata it seems that the index is a uint, so are the other bytes empty? or do we need to convert that byte array to number?

**Review by mikiquantum — APPROVED** [review]
LGTM, just a comment around the error format.

File: types/metadataV14.go (hunk 3 above)
Comment (cdamian) [reply types/metadataV14.go:163]: I was not able to find more information regarding those other 3 indexes, that's why locally I tested it with that and the first index seems to be the one that works. And tbh this is still confusing me. Probably we can take a look at that together to figure how it's supposed to be implemented here.

File: types/header.go (hunk 2 above)
Comment (cdamian) [reply types/header.go:73]: Yes, forgot to clean this up.

Comment (cdamian) [reply types/header.go]: Removed.

File: types/metadataV14.go (hunk 3 above)
Comment (cdamian) [reply types/metadataV14.go:163]: As discussed, let's keep this for the time being, since it's also similar to how it's done in `polkadot-js`.

**Review by mikiquantum — APPROVED** [review]
👍

**Pull request merged**